### DISCUSSION TITLE
vercel rewrites to static folder

### DISCUSSION THREAD

**Discussion by skene** [vercel/next.js#13521] [Help]
Hello,

Im trying to use vercel.json rewrites to direct urls to a specific file in my next js static folder with no luck. 

the **vercel.json** file
```
{
  "version": 2,
  "rewrites": [
    {
      "source": "/static/file.txt",
      "destination": "/src/public/folder/file.txt
    }
  ]
}
```

the repo structure

```
src
    components
    pages
    public
        folder
            file.txt
vercel.json
next.config.js
package.json
```
Do i need to be using the dist folder as the destination? If so, i am not seeing the file.txt being outputted to the dist static folder?

Thanks in advanced

**Comment by ijjk** [comment]
Hi, the `public` folder should not be nested under the `src` folder since it still needs to be at the top-level to be detected correctly. 

After moving the public folder to the top of your project you can also leverage the new experimental rewrites support in your `next.config.js` and they will automatically be applied when deploying on Vercel e.g.

```js
// next.config.js
module.exports = {
  experimental: {
    async rewrites() {
      return [
        {
          source: '/static/file.txt',
          destination: '/folder/file.txt'
        }
      ]
    }
  }
}
```
